File: opijae/programmers/짝지어제거하기.py
def solution(s):
    stack=[]
    for c in s:
        # stack의 마지막이랑 현재 문자가 같으면 제거
        if stack and c ==stack[-1]:
            stack.pop()
        else:
            stack.append(c)
    if len(stack)==0:
        return 1
    else:
        return 0

File: sally/programmers/week3/짝지어 제거하기.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int solution(string s) {
    int answer = 0;
    stack<int> st;
    int s_len = s.size();
    for(int i = 0; i < s_len; i++){
        if(st.empty()) {
            st.push(s[i]);
            continue;
        }
        
        if(st.top() == s[i]) st.pop();
        else st.push(s[i]);   
    }
    
    if(st.empty()) answer = 1;
    
    return answer;
}